task/github.py

```diff
@@ -169,14 +169,16 @@ def url(self):
         if not self._url:
             if not self._base:
                 netloc = os.environ.get("GITHUB_API", "https://api.github.com")
-                self._base = "{0}/repos/{1}/".format(netloc, self.repo)
+                self._base = "{0}/repos/{1}".format(netloc, self.repo)
 
             self._url = urllib.parse.urlparse(self._base)
 
         return self._url
 
     def qualify(self, resource):
-        return urllib.parse.urljoin(self.url.path, resource)
+        if resource is None:
+            return self.url.path
```

Comment:
That looks unrelated -- what's that for? Should it be a separate commit?

Reply:
ah yea, this should be a separate commit. Doing that :)

Reply:
Wait on second thought it shouldn't be, but it is kinda confusing. I can add a bit more explanation when squashing.

So `/repos/:owner/:repo` is also an endpoint, this has the `default_branch` value in it. And `qualify()` always appends an `/`, which breaks that particular endpoint.

This way `api.get()` will query the `/repos/:owner/:repo` endpoint.

Reply:
Thanks. I can't say that I fully understand this, but fine to land then with some extra explanation.

Reply:
So an api call to `/repos/cockpit-project/cockpit` returns an object; an api call to `/repos/cockpit-project/cockpit/` returns None

Reply:
Yes, I get that part, that's line 172. I meant the default fallback of "resource" here.